packages/amplify-graphql-api-construct/src/internal/data-source-config.ts

```diff
@@ -93,3 +99,65 @@ export const getDataSourceStrategiesProvider = (definition: IAmplifyGraphqlDefin
 
   return provider;
 };
+
+/**
+ * Creates a new schema by merging the individual schemas contained in the definitions, combining fields of the Query and Mutation types in
+ * individual definitions into a single combined definition. Adding directives to `Query` and `Mutation` types participating in a
+ * combination is not supported (the behavior is undefined whether those directives are migrated).
+ */
+export const schemaByMergingDefinitions = (definitions: IAmplifyGraphqlDefinition[]): string => {
+  const schema = definitions.map((def) => def.schema).join('\n');
+  const parsedSchema = parse(schema);
```

Comment:
When schemas in multiple definitions define custom `Query` types and we're joining and parsing them, doesn't the combined schema have multiple `Query` types?

Reply:
Yes, that is the point of this method. The `parse` method doesn't validate types, it just transforms the string schema into a GraphQL AST, which is what we operate on.